### PR TITLE
STOR-1055: Add CSIDriver verification check in library-go

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-1055
Enhancement: https://github.com/openshift/enhancements/blob/master/enhancements/storage/csi-inline-vol-security.md
Related PR's:
https://github.com/openshift/kubernetes/pull/1384
https://github.com/openshift/cluster-kube-apiserver-operator/pull/1385
https://github.com/openshift/csi-driver-shared-resource-operator/pull/59
/hold for those 3 PR's to merge first
/cc @openshift/storage

### CODE REVIEW

**Comment by dobsonj** [comment]
Tested this change with and without the label in https://github.com/openshift/csi-driver-shared-resource-operator/pull/60

**Comment by dobsonj** [comment]
/assign @soltysh
for `/approve`
this PR is only on hold until the other 3 PR's from the description merge first, so we do not enforce this check prematurely.

This should not be user-noticable change, it's meant to get CI to scream loudly at us if we try to ship a CSI driver without the new required label. Tested in https://github.com/openshift/csi-driver-shared-resource-operator/pull/60 to make sure it fails without the label and passes with the label.
/label docs-approved
/label px-approved
/label qe-approved

File: pkg/operator/resource/resourceapply/storage.go
Comment (soltysh) [inline pkg/operator/resource/resourceapply/storage.go:123]: Nit: I like when those are written like so:
```golang
if err = validateRequiredCSIDriverLabels(required); err != nil {
```
it makes them more readable and can potentially save from shadowing errors too.

File: pkg/operator/resource/resourceapply/storage.go
Comment (soltysh) [inline pkg/operator/resource/resourceapply/storage.go:189]: once you found it you could break the loop to save a few cycles, no?

**Review by soltysh — APPROVED** [review]
Minor nits that can be addressed in followup
/approve

File: pkg/operator/resource/resourceapply/storage.go
Comment (dobsonj) [reply pkg/operator/resource/resourceapply/storage.go:187]: yes, updated

File: pkg/operator/resource/resourceapply/storage.go
Comment (dobsonj) [reply pkg/operator/resource/resourceapply/storage.go]: done

**Comment by dobsonj** [comment]
/unhold
everything else is merged

**Pull request merged**
